server/db/schema/hostels.schema.ts

```diff
@@ -0,0 +1,45 @@
+import { relations } from 'drizzle-orm';
+import { pgTable, serial, text, timestamp, varchar } from 'drizzle-orm/pg-core';
+
+import { staff } from '.';
+
+export const hostels = pgTable('hostels', {
+  id: serial('id').primaryKey(),
+  address: text('address').notNull(),
+  name: varchar('name', { length: 256 }).notNull(),
+  email: varchar('email', { length: 256 }).notNull(),
+  telephone: varchar('telephone', { length: 13 }).notNull(),
+  alternateTelephone: varchar('alternate_telephone', { length: 13 }),
+  type: varchar('type', {
+    enum: ['boys', 'girls'],
+  }).notNull(),
+  createdAt: timestamp('created_at').defaultNow().notNull(),
+  updatedAt: timestamp('updated_at')
+    .$onUpdate(() => new Date())
+    .notNull(),
+});
+
+export const hostelRelations = relations(hostels, ({ many }) => ({
```

Comment:
Minor nit: `Relations` should just be a suffix, meaning the rest should be the exact name of the table.
```suggestion
export const hostelsRelations = relations(hostels, ({ many }) => ({
```